### PR TITLE
Apply dictionary date formats on csv downloads when in reference mode

### PR DESCRIPTION
**Pull request by janette**
ref: 21261

## QA Steps

- [ ] Set the data dictionary mode to 'reference' https://dkan.ddev.site/admin/dkan/data-dictionary/settings
- [ ] Create a data dictionary with a field of type date, set the format to %m/%d/%Y
- [ ] Create a dataset with a data file containing a date column that you used in your data dictionary, dates should be in mm/dd/YYYY format
- [ ] Use your data dictionary endpoint in the distribution describedBy field
- [ ] Save and run import queues
- [ ] Visit your dataset, adjust the url to call the streamed download, or use the link below replacing the uuid with your dataset uuid

https://dkan.ddev.site/api/1/datastore/query/505a1f80-7a95-4d9e-8eda-c57efdbf5c19/0/download?&format=csv

- [ ] Confirm the resulting csv has the dates in the DD format, not ISO

### CODE REVIEW

File: modules/datastore/src/Service/ResourceProcessor/DictionaryEnforcer.php [hunk 1]
```diff
@@ -124,14 +124,25 @@ public function applyDictionary(RootedJsonData $dictionary, string $datastore_ta
    *
    *  {@inheritdoc}
    */
-  public function returnDataDictionaryFields() {
-    // Get DD is mode.
+  public function returnDataDictionaryFields($id = NULL) {
```
Comment (dafeder) [inline modules/datastore/src/Service/ResourceProcessor/DictionaryEnforcer.php]: This function should have a docbloc, the method is unique to this class and is not defined in the interface.

File: modules/datastore/src/DatastoreService.php [hunk 2]
```diff
@@ -217,8 +217,8 @@ public function getImportService(DataResource $resource): ImportService {
   /**
    * Returns the Data Dictionary fields.
    */
-  public function getDataDictionaryFields() {
-    return $this->dictionaryEnforcer->returnDataDictionaryFields();
+  public function getDataDictionaryFields($id = NULL) {
```
Comment (dafeder) [inline modules/datastore/src/DatastoreService.php]: Let's document this method properly with docbloc and type/return hints

**Review by dafeder — CHANGES_REQUESTED** [review]

File: modules/datastore/src/DatastoreService.php (hunk 2 above)
Comment (janette) [reply modules/datastore/src/DatastoreService.php]: Good call thanks

File: modules/datastore/src/Service/ResourceProcessor/DictionaryEnforcer.php (hunk 1 above)
Comment (janette) [reply modules/datastore/src/Service/ResourceProcessor/DictionaryEnforcer.php]: Right again

**Review by dafeder — APPROVED** [review]

**Pull request merged**